rtl/include/eclhelper.hpp

```diff
@@ -3019,6 +3019,30 @@ class XmlChildIterator
 constexpr unsigned daliResultOutputMax = 2000; // MB
 constexpr unsigned futureResultOutputMax = 100; // MB
 
+//------------------------------------------------------------------------------------------------
+
+// RAII class to start and stop a section timer
+class SectionTimerBlock
+{
+private:
+    Linked<ISectionTimer> timer;
```

Comment:
I think better if this is a ISectionTimer * (i.e. not linked).  The lifetime of the timer must be longer than the block that is using it, and it will avoid an atomic inc/dec.